### PR TITLE
Adjust aml impact on tables shown

### LINKED ISSUE #132: Only remove table from current view if it does not exist anymore (AML)
When deleting a table definition from AML, it systematically hides the table in the view even if it still exists.

For example, I use a custom AML source 'default' to update an existing `account` table with a `title` column:
image
(account table is also defined elsewhere as there is an `id` column as well)

If I remove the `account` table definition from the 'default' source POV, the table will be hidden from the view. UX wise I feel it would be better if table were hidden if there is no selected configuration left to display them.

### PR DESCRIPTION
**Pull request by alxckn**
Fixes #132

### CODE REVIEW

File: frontend/src/PagesComponents/Organization_/Project_/Components/AmlSidebar.elm [hunk 1]
```diff
@@ -171,6 +189,23 @@ contentStr source =
...
+    case currentSourceId of
+        Nothing ->
+            Set.empty
+
+        Just id ->
+            let
+                otherSources : List Source
+                otherSources =
+                    erd
+                        |> Maybe.map (.sources >> List.filterNot (\s -> s.id == id))
```
Comment (loicknuchel) [inline frontend/src/PagesComponents/Organization_/Project_/Components/AmlSidebar.elm:203]:
```suggestion
                    erd |> Maybe.mapOrElse .sources [] |> List.filterNot (\s -> s.id == id)
```

File: frontend/src/PagesComponents/Organization_/Project_/Components/AmlSidebar.elm [hunk 2]
```diff
@@ -171,6 +189,23 @@ contentStr source =
...
+                        |> Maybe.withDefault []
+            in
+            otherSources |> List.concatMap (\s -> s.tables |> Dict.values) |> List.map .id |> Set.fromList
```
[the remark this answers is not in the document]
Comment (loicknuchel) [reply frontend/src/PagesComponents/Organization_/Project_/Components/AmlSidebar.elm:206]: Looking at this comment and the previous one, maybe they can be combined with:
```elm
erd
  |> Maybe.mapOrElse .sources [] 
  |> List.filterNot (\s -> s.id == id)
  |> List.concatMap (.tables >> Dict.keys)
  |> Set.fromList
```

File: frontend/src/PagesComponents/Organization_/Project_/Components/AmlSidebar.elm [hunk 3]
```diff
@@ -109,17 +118,26 @@ updateSource now source input model =
...
         toHide : List TableId
         toHide =
-            removed |> List.map .id
+            trulyRemoved |> List.map .id
 
-        updated : List ( TableId, List ColumnName )
+        updated : List Table
         updated =
-            bothPresent |> List.filter (\( t1, t2 ) -> t1 /= t2) |> List.map (\( _, t ) -> ( t.id, t.columns |> Dict.keys ))
+            bothPresent |> List.filter (\( t1, t2 ) -> t1 /= t2) |> List.map (\( _, t ) -> t)
```
Comment (loicknuchel) [inline frontend/src/PagesComponents/Organization_/Project_/Components/AmlSidebar.elm:135]:
```suggestion
            bothPresent |> List.filter (\( t1, t2 ) -> t1 /= t2) |> List.map Tuple.second
```

**Review by loicknuchel — COMMENTED** [review]
Looks good, thanks!
Mostly just syntax suggestions ;)

File: frontend/src/PagesComponents/Organization_/Project_/Components/AmlSidebar.elm [hunk 4]
```diff
@@ -109,17 +118,26 @@ updateSource now source input model =
         ( removed, bothPresent, added ) =
             List.diff .id (source.tables |> Dict.values) (parsed.tables |> Dict.values)
 
+        tableIdsOutsideSource : Set TableId
+        tableIdsOutsideSource =
```
[the remark this answers is not in the document]
Comment (alxckn) [reply frontend/src/PagesComponents/Organization_/Project_/Components/AmlSidebar.elm:122]: Yes sorry about that, creating the cache and not using it :open_mouth:

**Review by loicknuchel — APPROVED** [review]
Looks good.
Will just take some time to run it before merge but :+1:

**Comment by alxckn** [comment]
Thanks for the review @loicknuchel !

**Pull request merged**